cmd/node/stats.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	pb "adaptodb/pkg/proto/proto"
+	"adaptodb/pkg/sm"
+
+	"github.com/lni/dragonboat/v3"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+
+type NodeStatsServer struct {
+	pb.UnimplementedNodeStatsServer
+	mu sync.Mutex
+	entries int64
+	successfulRequests int64
+	failedRequests int64
+	nh *dragonboat.NodeHost	
+	clusterID uint64
+	lastResetTime time.Time
+}
+
+func NewNodeStatsServer(nh *dragonboat.NodeHost, clusterID uint64) *NodeStatsServer {
```

Comment:
Is this method to create a new node stats server? Maybe `createNodeStatsServer` is a more suitable method name? Nit pick though

Reply:
Also is this a singleton?

Reply:
The `NewXxxx` is the naming convention for a "constructor". The `New` is capitalized because they are public